src/world.rs

```diff
@@ -130,19 +130,15 @@ impl World {
             std::process::exit(0);
         }
 
-
-        if is_mouse_button_down(macroquad::input::MouseButton::Left) {
+        if is_mouse_button_released(macroquad::input::MouseButton::Left) {
             let mouse_position: Vec2 = macroquad::input::mouse_position().into();
             let primary_ball = self.primary_ball_mut();
-            let d = primary_ball.position().distance(mouse_position);
-            let dx = primary_ball.position().x - mouse_position.x;
-            let dy = primary_ball.position().y - mouse_position.y;
-
-            let vx = (dx / d).sin();
-            let vy = (dy / d).sin();
-            let v = Vec2::new(vx, vy);
-            dbg!(v);
-            primary_ball.push(-v * 2.0);
+            let d = primary_ball.position().distance(mouse_position).max(0.001);
```

Comment:
is `.max(0.001)` really needed there?
```suggestion
            let d = primary_ball.position().distance(mouse_position);
```

Reply:
let d = primary_ball.position().distance(mouse_position).max(0.001);
            let v = (*primary_ball.position() - mouse_position) / d;
yes, to ensure d != 0